File: frontend/src/api/services/groupService.js
import api from "../api";

const groupService = {
  getAllGroups: () => api.get("/groups").then((res) => res.data),
  getGroupById: (groupId) =>
    api.get(`groups/${groupId}`).then((res) => res.data),
  createGroup: (group) => api.post(`groups`, group).then((res) => res.data),
};

export default groupService;

File: backend/src/main/java/io/plan/mate/expense/tracker/backend/group/service/GroupService.java
package io.plan.mate.expense.tracker.backend.group.service;

import io.plan.mate.expense.tracker.backend.commons.service.dto.PagedResponse;
import io.plan.mate.expense.tracker.backend.group.service.dto.GroupDto;
import io.plan.mate.expense.tracker.backend.group.controller.payload.request.CreateGroupRequest;

import org.springframework.data.domain.Pageable;

public interface GroupService {

    GroupDto createGroup(CreateGroupRequest createGroupRequest);

    GroupDto getGroupById(Long id);

    PagedResponse<GroupDto> getAllGroups(Pageable pageable);

    void deleteGroup(Long id);
}
